testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpServerTest.groovy

```diff
@@ -254,9 +262,12 @@ abstract class HttpServerTest<SERVER> extends InstrumentationSpecification imple
     def response = client.newCall(request).execute()
 
     expect:
-    response.code() == REDIRECT.status
-    response.header("location") == REDIRECT.body ||
-      response.header("location") == "${address.resolve(REDIRECT.body)}"
+    response.withCloseable {
+      assert response.code() == REDIRECT.status
```

Comment:
Digging more to try to understand what's going on, I found that most of our tests were actually closing, just not these header-only ones

https://github.com/square/okhttp/blob/parent-3.14.9/okhttp/src/main/java/okhttp3/ResponseBody.java#L48

That being said it's cleaner code to always use closeable without relying on this magic close-by-`string()` invocation.

But given that, I can only hypothesis that not closing in redirect test had an adverse affect on the error test. Even so, I think it's only supposed to be about whether a pooled buffer is returned to the buffer pool, and even if there happened to be buffer corruption, I'd expect it to have the word `/redirected` and not `controller error`. So still not exactly sure why tests would have started to correctly fail after this change but won't let Murphy affect my Saturday much more :)

Reply:
Smoke tests had the same issue, there I added
```
    cleanup:
    response?.close()
```

I believe that in this case we were just bitten by magic of spock. Considering that in
```
  def "test a"() {
    expect:
    1 == 1
    1 == 0
  }

  def "test b"() {
    expect:
    1 == 1
    if (1 == 1) {
      1 == 0
    }
  }
```
a fails but b passes I suppose you just enabled an assertion that was previously ignored :)